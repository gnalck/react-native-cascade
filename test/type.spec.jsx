import React from 'react';
import 'react-native';
import renderer from 'react-test-renderer';
import {shallow} from 'enzyme';
import Type from './Type';


it('matches the snapshot for a string type attribute', () => {
  const tree = renderer
    .create(<Type />)
    .toJSON();
  expect(tree).toMatchSnapshot();
});

it('creates a style node with appropriate style for a type attribute', () => {
  const com = shallow(<Type />);
  const hasStylesNode = com.find({'test-id': 'base'});
  expect(JSON.stringify(hasStylesNode.prop('style'))).toBe(JSON.stringify({
    color: 'salmon',
    backgroundColor: 'chartreuse',
  }));
});
