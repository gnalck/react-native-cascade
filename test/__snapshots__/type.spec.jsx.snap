// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches the snapshot for a string type attribute 1`] = `
<View>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Object {
        "backgroundColor": "chartreuse",
        "color": "salmon",
      }
    }
    test-id="base"
  >
    Hello
  </Text>
  <View
    test-id="unmatched"
  />
</View>
`;
